File: 프로그래머스/0/181922. 수열과 구간 쿼리 4/수열과 구간 쿼리 4.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries) {
    for (auto v : queries) {
        for (int i = v[0]; i <= v[1]; i++) {
            if (i % v[2] == 0) {
                arr[i]++;
            }
        }
    }

    return arr;
}

File: 프로그래머스/0/181922. 수열과 구간 쿼리 4/수열과 구간 쿼리 4.swift
import Foundation

func solution(_ arr:[Int], _ queries:[[Int]]) -> [Int] {
    
    var result = arr
    
    queries.forEach {
        for index in $0[0]...$0[1] {
            if index.isMultiple(of: $0[2]) { result[index] += 1}
        }
    }
    
    return result
}